dist: trusty
sudo: false
language: python
cache: pip
services:
  - redis-server
  - postgresql
addons:
  postgresql: "9.6"
  apt:
    packages:
    - postgresql-9.6-postgis-2.3
    - libgdal-dev
python:
  - 3.5
  - 3.6
env:
  - DJANGO=1.11
  - DJANGO=2.0
matrix:
  include:
    - python: 3.5
      env: DJANGO=1.11
    - python: 3.5
      env: DJANGO=2.0
    - python: 3.6
      env: DJANGO=1.11
    - python: 3.6
      env: DJANGO=2.0
  allow_failures:
    - python: 3.5
      env: DJANGO=2.0
    - python: 3.6
      env: DJANGO=2.0
install:
  - pip install -r requirements.txt
before_script:
  # check gis
  - gdal-config --version
  - gdal-config --cflags
  - psql -U postgres -c "create extension postgis"
  # set up postgresql
  - psql -U postgres -c "create role cacheops login superuser"
  # postgis django backend requires these to exist
  - psql -U postgres -c "create database cacheops"
  - psql -U postgres -c "create database cacheops_slave"
script:
  - GDAL_VERSION=1.10 ./manage.py test --settings=pasportaservo.settings.testing
